src/API/auth/auth.controller.js

```diff
@@ -37,34 +44,46 @@ function register() {
                 })
                 .catch(next);
         },
-        passport.authenticate('local', {session: false}),
-        (req, res) => {
-            res.json({
-                user: req.user,
-                token: authService.generateToken(req.user._id)
-            });
-        }
+        createPassportAuthenticateMiddleware('local'),
+        generateTokenControllerAction
     );
 }
 
 function login() {
     return composeMiddleware.compose(
-        (reg, res, next) => {
-            reg.params = reg.body;
-            next();
-        },
-        passport.authenticate('local', {session: false}),
-        (req, res) => {
-            res.json({
-                user: req.user,
-                token: authService.generateToken(req.user._id)
-            });
-        }
+        bodyToParamsMiddleware,
+        createPassportAuthenticateMiddleware('local'),
+        generateTokenControllerAction
+    );
+}
+
+function facebookLoginWithAccessToken() {
```

Comment:
I think that ```WithAccessToken``` suffix is redundant, in the reason that we have no opposite method called ```facebookLoginWithoutAccessToken```